aspnetcore/signalr/httpcontext.md

```diff
@@ -0,0 +1,25 @@
+---
+title: IHttpContextAccessor/HttpContext in ASP.NET Core SignalR
+author: guardrex
+description: Learn about IHttpContextAccessor and HttpContext in ASP.NET Core SignalR.
+monikerRange: '>= aspnetcore-3.1'
+ms.author: riande
+ms.custom: mvc
+ms.date: 01/30/2025
+uid: signalr/httpcontext
+---
+# `IHttpContextAccessor`/`HttpContext` in ASP.NET Core SignalR
+
+[!INCLUDE[](~/includes/not-latest-version-without-not-supported-content.md)]
+
+<xref:Microsoft.AspNetCore.Http.IHttpContextAccessor>/<xref:Microsoft.AspNetCore.Http.HttpContext> generally should be avoided with SignalR because a valid <xref:Microsoft.AspNetCore.Http.HttpContext> isn't always available. In most cases, the context doesn't exist (`null`).
+
+Even when an <xref:Microsoft.AspNetCore.Http.HttpContext> instance is available, the context is dependent on the transport:
+
+* WebSockets receives a single context as the result of the initial handshake.
+* Long polling receives a new context per client "poll" request.
+* A SignalR service receives a mocked/faked/shim context.
+
+When in the hub, <xref:Microsoft.AspNetCore.SignalR.GetHttpContextExtensions.GetHttpContext%2A?displayProperty=nameWithType> returns the <xref:Microsoft.AspNetCore.Http.HttpContext> for the connection or `null` if the connection isn't associated with an HTTP request. Use this method to get HTTP connection information, such as HTTP headers and query strings. For more information, see <xref:signalr/hubs#the-context-object>.
```

Comment:
Maybe this should be more explicit about using `GetHttpContext` instead of `IHttpContextAccessor` because we (SignalR http layer) create a fake context that will always exist for http connections. Technically it won't have everything that the original `HttpContext` has since we manually populate it, but it is more guaranteed to exist.

Reply:
Try the latest commit ... I roll those facets into the existing sentence.